### ISSUE TITLE
VGG19 model fails to load with EOFError

### ISSUE THREAD

**Issue by AdarshKumar712** [FluxML/Metalhead.jl#58]
I am trying to load the VGG19 model on CPU, however in the end it throws me an End of File Error.
It shows following error:
```
ERROR: EOFError: read end of file
Stacktrace:
 [1] read at ./iostream.jl:350 [inlined]
 [2] read(::IOStream, ::Type{BSON.BSONType}) at ./Enums.jl:22
 [3] parse_doc(::IOStream) at /home/blackforest/.julia/packages/BSON/lY9i8/src/read.jl:51
 [4] parse_tag(::IOStream, ::BSON.BSONType) at /home/blackforest/.julia/packages/BSON/lY9i8/src/read.jl:15
 [5] parse_doc(::IOStream) at /home/blackforest/.julia/packages/BSON/lY9i8/src/read.jl:53
 [6] parse at /home/blackforest/.julia/packages/BSON/lY9i8/src/read.jl:101 [inlined]
 [7] #open#271(::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}, ::typeof(open), ::typeof(BSON.parse), ::String) at ./io.jl:298
 [8] open at ./io.jl:296 [inlined]
 [9] parse at /home/blackforest/.julia/packages/BSON/lY9i8/src/read.jl:102 [inlined]
 [10] load at /home/blackforest/.julia/packages/BSON/lY9i8/src/read.jl:104 [inlined]
 [11] weights(::String) at /home/blackforest/.julia/packages/Metalhead/RZn9O/src/utils.jl:21
 [12] vgg19_layers() at /home/blackforest/.julia/packages/Metalhead/RZn9O/src/vgg19.jl:2
 [13] VGG19() at /home/blackforest/.julia/packages/Metalhead/RZn9O/src/vgg19.jl:39
 [14] top-level scope at REPL[2]:1
```

**Comment by avik-pal** [comment]
@AdarshKumar712 looks like there was some issue in downloading and the file is incomplete. Maybe we should use some sort of verification to make ensure this.

**Comment by AdarshKumar712** [comment]
I also think that weights are not downloaded properly. Exactly what kind of verification can be done here? I tried running the Pkg.test("Metalhead"). It showed the same error for the VGG19 model.
However, all other models work fine except for VGG19.

**Comment by avik-pal** [comment]
@AdarshKumar712 I didn't mean CI testing. I was referring to using checksum to validate if the file is corrupt and if so simply re-download it.
[Here](https://www.howtogeek.com/363735/what-is-a-checksum-and-why-should-you-care/) is an article about checksum.

**Comment by AdarshKumar712** [comment]
Yes, the issue got solved after I deleted and re-downloaded vgg19.bson. But for the future, where to get the FileHash for these bson files of weights to use for checksum?

**Comment by avik-pal** [comment]
I don't think these are present as of now. Generating them might be a good idea though. @dhairyagandhi96 Thoughts on this?

**Comment by ven-k** [comment]
I ran into the same problem. I deleted, garbage-collected and re-downloaded the whole package, multiple times. The problem persists. Is there a way I can download only the weights and then edit the joinpath in weights function? (which is given by [11] in the error message)

**Comment by AdarshKumar712** [comment]
Actually you don't need to delete the whole package. Just delete the bson file of that model from deps folder under .julia/packages/Metalhead/.../ , and then re-download the file.

Manually, you can download the weight file for vgg19 from "https://github.com/FluxML/Metalhead.jl/releases/download/Models/vgg19.bson" and others from "https://github.com/FluxML/Metalhead.jl/releases/download/v0.1.1/(model_name).bson" and then save the downloaded file to deps folder(is not present, you need to create one) under .julia/packages/Metalhead/.../. I think then, you won't need to edit the joinpath the weights function

**Comment by AdarshKumar712** [comment]
@avik-pal Can DataDeps.jl be used here to manage downloading of model weights?

**Comment by avik-pal** [comment]
I not very familiar with DataDeps.jl, but after a quick glance through its docs, it could be used for this.

**Comment by ven-k** [comment]
Thanks, @AdarshKumar712. It worked!

**Comment by ven-k** [comment]
Thanks, @AdarshKumar712. It helped.

**Comment by clad26** [comment]
Hmmm.... shouldn't we implement a check in Metalhead, and re-download the bson file, if its corrupt(i.e, it encounters the EOF error)?

**Comment by DhairyaLGandhi** [comment]
We can check the sha and do it that way. Do we store that already?

**Comment by clad26** [comment]
I use try-catch to get around this

try
   statement
catch
   download("link_to_vgg19", path_to_vgg19)
   statement
end
